### PR TITLE
Show `description` on fields without a `label`

### PR DESCRIPTION
**Pull request by michaeltaranto**
Previously, a `FieldLabel` without a `label` would return nothing. However, now that form fields can opt for [indirect or hidden labels] (via `aria-label` or `aria-labelledby`), the `description` should still be shown if provided.

> Note: The `secondaryLabel` and `tertiaryLabel` remain conditional based on the presence of a `label`. This is due to their location in the layout being anchored off the `label`.

**EXAMPLE USAGE:**
```jsx
<FieldLabel
  description="Description now visible without label"
/>
```

[indirect or hidden labels]: https://seek-oss.github.io/braid-design-system/components/TextField#indirect-or-hidden-field-labels

### CODE REVIEW

File: .changeset/great-gorillas-live.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - RadioGroup
+  - TextField
+  - Textarea
+---
+
+Show `description` on form fields when no `label` provided
+
+Previously, a `FieldLabel` without a `label` would return nothing. However, now that form fields can opt for [indirect or hidden labels] (via `aria-label` or `aria-labelledby`), the `description` should still be shown.
+
+> Note: The `secondaryLabel` and `tertiaryLabel` remain conditional based on the presence of a `label`. This is due to the their location in the layout being anchored off the `label`.
```
Comment (jahredhope) [inline .changeset/great-gorillas-live.md:21]:
```suggestion
> Note: The `secondaryLabel` and `tertiaryLabel` remain conditional based on the presence of a `label`. This is due to their location in the layout being anchored off the `label`.
```

File: .changeset/great-gorillas-live.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  - MonthPicker
+  - PasswordField
+  - RadioGroup
+  - TextField
+  - Textarea
+---
+
+Show `description` on form fields when no `label` provided
+
+Previously, a `FieldLabel` without a `label` would return nothing. However, now that form fields can opt for [indirect or hidden labels] (via `aria-label` or `aria-labelledby`), the `description` should still be shown.
```
Comment (michaeltaranto) [inline .changeset/great-gorillas-live.md:19]:
```suggestion
Previously, a `FieldLabel` without a `label` would return nothing. However, now that form fields can opt for [indirect or hidden labels] (via `aria-label` or `aria-labelledby`), the `description` should still be shown if provided.
```

File: packages/braid-design-system/src/lib/components/Textarea/Textarea.screenshots.tsx [hunk 3]
```diff
@@ -64,6 +64,19 @@ export const screenshots: ComponentScreenshot = {
         <Textarea id={id} value="" onChange={handler} aria-label="Label" />
       ),
     },
+    {
+      label: 'Textarea with a description and no visual label ',
```
Comment (askoufis) [inline packages/braid-design-system/src/lib/components/Textarea/Textarea.screenshots.tsx:68]:
```suggestion
      label: 'Textarea with a description and no visual label',
```

File: packages/braid-design-system/src/lib/components/TextField/TextField.screenshots.tsx [hunk 4]
```diff
@@ -129,6 +129,19 @@ export const screenshots: ComponentScreenshot = {
         <TextField aria-label="Label" id={id} value="" onChange={handler} />
       ),
     },
+    {
+      label: 'TextField with a description and no visual label ',
```
Comment (askoufis) [inline packages/braid-design-system/src/lib/components/TextField/TextField.screenshots.tsx:133]:
```suggestion
      label: 'TextField with a description and no visual label',
```

File: packages/braid-design-system/src/lib/components/private/Field/Field.tsx [hunk 5]
```diff
@@ -25,19 +25,16 @@ export type FieldLabelVariant =
...
       'aria-label': string;
       secondaryLabel?: never;
       tertiaryLabel?: never;
-      description?: never;
     }
   | {
       label: FieldLabelProps['label'];
       secondaryLabel?: FieldLabelProps['secondaryLabel'];
       tertiaryLabel?: FieldLabelProps['tertiaryLabel'];
-      description?: FieldLabelProps['description'];
```
Comment (askoufis) [inline packages/braid-design-system/src/lib/components/private/Field/Field.tsx:40]: Out of scope for this PR, but would there be any value in displaying these prop type variants more explicitly on the component docs pages? Currently `label`, `aria-label` and `aria-labelledby` are all displayed as `Required`, when in reality only one of them is required. I'm guessing this an artifact of how the type contract serializer compresses these union types.

image

Comment (michaeltaranto) [reply packages/braid-design-system/src/lib/components/private/Field/Field.tsx:40]: Yeah we can pick this up, there are current limitations to the type extraction script.

**Review by askoufis — APPROVED** [review]

**Pull request merged**